src/main/java/com/gooddata/retry/RetryableRestTemplate.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2007-2017, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.retry;
+
+import com.gooddata.GoodDataRestException;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.web.client.RequestCallback;
+import org.springframework.web.client.ResponseExtractor;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+/**
+ * REST template with retry ability. Its behavior is described by given strategy and retry template.
+ */
+public class RetryableRestTemplate extends RestTemplate {
```

Comment:
Hmm it's kind of weird mixin Retryable and FailOver....let's discuss in person

Reply:
ok :-)